addons/hr_timesheet/controllers/portal.py

```diff
@@ -139,8 +139,14 @@ def get_timesheets():
 
                 else:
                     time_data = Timesheet_sudo._read_group(domain, [field], ['unit_amount:sum'])
-                    mapped_time = {field.id: unit_amount for field, unit_amount in time_data}
-                    grouped_timesheets = [(Timesheet_sudo.concat(*g), mapped_time[k.id]) for k, g in groupbyelem(timesheets, itemgetter(field))]
+                    mapped_time = {
+                        field.id: (unit_amount, field.allocated_hours if hasattr(field, 'allocated_hours') else 0)
```

Comment:
```suggestion
                        field.id: (unit_amount, field.allocated_hours if  groupby  in ['task_id', ..] else 0)
```
When timesheets are grouped by 'task,' 'project,' and 'parent task,' the total allocated hours should be displayed. We should directly use the groupby attribute. 

If another model contains the allocated_hours attribute, the total allocated hours will be displayed because [hasattr](https://docs.python.org/3/library/functions.html#hasattr) checks if an object has the specified attribute, returning True if it is present and False otherwise.